lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -80,16 +88,17 @@ class _Visitor extends SimpleAstVisitor {
 
   void _visitInvocationExpression(
       InvocationExpression node, FunctionExpression nodeToLint) {
-    if (nodeToLint.parameters.length != node.argumentList.length) {
+    if (nodeToLint.parameters.parameters.length !=
+        node.argumentList.arguments.length) {
```

Comment:
I added a commented test with a TODO statement, I could not reproduce the bug without generics